internal/vds/attribute.cpp

```diff
@@ -51,6 +51,29 @@ float Mean::compute(Mean::InputIt begin, Mean::InputIt end) noexcept (false) {
     return sum / this->vsize;
 }
 
+
+float Median::compute(
+    const Median::InputIt begin,
+    const Median::InputIt end
+) noexcept (false) {
+    /*
+    The std::nth_element gives the middle element and it ensures that all
+    elements preceding the middle are less than the middle value. For the even
+    number of elements, we use std::max_element to find the largest element
+    preceding the middle element to calculate the average.
+    */
+    auto temp = std::vector<double>(begin, end);
+    const auto middle_right = temp.begin() + vsize / 2;
+    std::nth_element(temp.begin(), middle_right, temp.end());
+    if (vsize % 2 == 0) {
+        const auto end_left = std::max_element(temp.begin(), middle_right);
```

Comment:
The logic is to find the maximum value from the left half of the vector because that will be the end value in the case of a sorted vector. But I can change this to max_left which is a more suitable name.